Feature/speech/weather api

henrikschnor opened the pull request:
The questions "how is the weather?" and "where are you?" were answered with hardcoded data (e.g. it's always "sunny"). I implemented the use of free accessible APIs (IP location + Yahoo weather) to answer those questions with actual data. This pull request refers to #79

henrikschnor committed:
Added a class for easy retrieval of weather information

henrikschnor committed:
Use a public API for IP info retrieval

henrikschnor committed:
Answer the weather and location question with data from public APIs

henrikschnor committed:
Fixed quality review warnings

PatrickNa commented:
We have tested the code on the robot today and it held what it promised.

alex-mitrevski commented on mdr_speech/mdr_question_answering/common/src/ip_info/ip_info.py:
```diff
@@ -0,0 +1,30 @@
+import requests
+
+
+class IPInfo:
```
Please inherit from `object` explicitly.

alex-mitrevski commented on mdr_speech/mdr_question_answering/common/src/ip_info/ip_info.py:
```diff
@@ -0,0 +1,30 @@
+        try:
+            return requests.get("http://ip-api.com/json").json()
+        except Exception:
+            # Something went wrong
```
It would be useful to at least write the exception message to the console here.

alex-mitrevski commented on mdr_speech/mdr_question_answering/common/src/weather_api/weather_api.py:
```diff
@@ -0,0 +1,81 @@
+import urllib
+
+
+class WeatherApi:
```
Please inherit from `object`

alex-mitrevski commented on mdr_speech/mdr_question_answering/common/src/weather_api/weather_api.py:
```diff
@@ -0,0 +1,81 @@
+                # Location not found
+                return None
+        except Exception:
+            # Something went wrong
```
Same here; write the exception message to the console

alex-mitrevski commented on mdr_speech/mdr_question_answering/common/src/weather_api/weather_api.py:
```diff
@@ -0,0 +1,81 @@
+        """
+        Converts a given temperature from fahrenheit to celsius
+        """
+        return (temperature - 32) * (5/9)
```
Unfortunately, we generally still use Python 2 for ROS-related activities and in Python 2, the division will result in the integer 0, which is not what you intend here (on the other hand, it will result in a floating-point value in Python 3). Change the division to `(5. / 9.)` to resolve the compatibility issue.

PatrickNa replied:
Valid request. Thanks for looking that detailed into it!

henrikschnor committed:
Fixed temperature conversion and added exception logging

henrikschnor committed:
Build proper phrases for weather conditions that are not adjectives

henrikschnor commented:
@alex-mitrevski Thanks for reviewing the code! I implemented your comments and added another commit for building proper phrases when the condition is not an adjective, so we don't get "it's scattered showers ..." (occured yesterday).

alex-mitrevski commented:
OK, this is now ready to be merged. An official welcome to the contributors club for @henrikschnor
